libs/ml/tests/unit/ops/activations/softmax.cpp

```diff
@@ -50,7 +50,8 @@ TYPED_TEST(SoftmaxTest, forward_test)
   op.Forward({std::make_shared<const TensorType>(data)}, prediction);
 
   // test correct values
-  ASSERT_TRUE(prediction.AllClose(gt, DataType{1e-5f}, DataType{1e-5f}));
+  ASSERT_TRUE(prediction.AllClose(gt, fetch::math::AsType<DataType>(1e-5f),
```

Comment:
more string constructions? I presume it'll still pass; in any case shouldn't we be using function tolerance here rather than 1e-5?

Reply:
similarly throughout this file